common/convert/config.go

```diff
@@ -0,0 +1,118 @@
+package convert
+
+import (
+	"encoding/json"
+	"strings"
+
+	dsc "github.com/aserto-dev/go-directory/aserto/directory/common/v3"
+	"github.com/aserto-dev/scim/common"
+	"github.com/aserto-dev/scim/common/config"
+	"github.com/pkg/errors"
+)
+
+type TemplateName int
+
+const (
+	Users TemplateName = iota
+	UsersGroups
+	UsersGroupsRoles
+)
+
+var ErrInvalidConfig = errors.New("invalid config")
+
+func (t TemplateName) String() string {
+	switch t {
+	case Users:
+		return "users"
+	case UsersGroups:
+		return "users-groups"
+	case UsersGroupsRoles:
+		return "users-groups-roles"
+	}
+
+	return "unknown"
+}
+
+type TransformConfig struct {
+	*config.Config
+	template           TemplateName
+	IdentityObjectType string `json:"identity_object_type,omitempty"`
+	IdentityRelation   string `json:"identity_relation,omitempty"`
+}
+
+func NewTransformConfig(cfg *config.Config) (*TransformConfig, error) {
+	template := Users
+
+	if cfg.Group != nil {
+		template = UsersGroups
+		if cfg.Role != nil {
+			template = UsersGroupsRoles
+		}
+	}
+
+	object, relation, found := strings.Cut(cfg.User.IdentityRelation, "#")
+	if !found {
+		return nil, errors.Wrap(ErrInvalidConfig, "identity relation must be in the format object#relation")
+	}
+
+	if object != cfg.User.IdentityObjectType && object != cfg.User.ObjectType {
+		return nil, errors.Wrapf(ErrInvalidConfig, "identity relation object type [%s] doesn't match user or identity type", object)
+	}
+
+	if relation == "" {
+		return nil, errors.Wrap(ErrInvalidConfig, "identity relation is required")
+	}
+
+	return &TransformConfig{
+		Config:             cfg,
+		template:           template,
+		IdentityObjectType: object,
+		IdentityRelation:   relation,
+	}, nil
+}
+
+func (c *TransformConfig) HasGroups() bool {
```

Comment:
If you made this a function on `config.Config` then you could use it in `NewTransformConfig`.